File: src/pedometer.c
#include <stdint.h>
#include <stdio.h>
#include <time.h>

struct PedoEntry {
  uint32_t timestamp;
  uint16_t steps;
} __attribute__((packed));

int main() {
  FILE *pedo = fopen("data/Pedometer.dat", "rb");
  if (!pedo) {
    perror("Failed to open pedometer.dat");
    return 1;
  }

  struct PedoEntry entry;
  while (fread(&entry, sizeof(struct PedoEntry), 1, pedo) == 1) {
    if (entry.timestamp != 0) {
      uint32_t timestamp = entry.timestamp;
      time_t ts = (time_t)timestamp;
      struct tm *tm_info = gmtime(&ts);
      char buffer[26];
      strftime(buffer, sizeof(buffer), "%Y-%m-%d", tm_info);
      printf("Timestamp: %u, Date: %s, Steps: %u\n", timestamp, buffer,
             entry.steps);
    }
  }

  fclose(pedo);
  return 0;
}

File: src/pedometer.py
import struct
from datetime import datetime

RECORD_SIZE = 6  # u32 (4 bytes) + u16 (2 bytes)

with open("data/Pedometer.dat", "rb") as f:
    record_num = 0
    while True:
        data = f.read(RECORD_SIZE)
        if len(data) < RECORD_SIZE:
            break

        timestamp, steps = struct.unpack("<IH", data)
        date = datetime.fromtimestamp(timestamp)

        # if date.year in range(2016, 2026):
        print(f"Record {record_num}: Date {date}, Steps {steps}")

        record_num += 1
